src/datreant/cli.py

```diff
@@ -0,0 +1,396 @@
+              help="return relative or absolute paths")
+@click.option("--json", '-j', 'json_', is_flag=True,
+              help="return results in JSON format")
+@click.option("--tags", '-t', cls=OptionEatAll,
+              help="list of tags or search string")
+@click.option(
+    "--categories",
+    '-c',
+    cls=OptionEatAll,
+    help="list of categories as key-value pairs separated by a colon ':'",
+)
+def get(dirs, relpath, json_, tags, categories):
+    """Filter treants that strictly match tags and/or categories
+
+    """
+    dirs = _handle_stdin(dirs)
+
+    dirs = dtr.Bundle(dirs, ignore=True)
+
+    res = _get(dirs, tags, categories)
+
+    if json_:
+        res = res.map(print_treant, detail=True)
+        if res:
+            click.echo(json.dumps(res, indent=4))
+    else:
+        click.echo("\n".join(res.abspaths))
+
+
+@cli.command()
+@click.argument("dirs", metavar='<dir(s)>', nargs=-1)
+@click.option("--all/--any", 'all_', default=True,
+              help="results for tags present in ALL or ANY treants")
+@click.option("--json", '-j', 'json_', is_flag=True,
+              help="return results in JSON format")
+@click.option("--has", cls=OptionEatAll,
+              help=("list of tags; for each returns true if present,"
+                    " false if absent"))
+def tags(dirs, all_, json_, has):
+    """Show treant tags"""
+    dirs = _handle_stdin(dirs)
+
+    # get all existing Treants among dirs
+    dirs = dtr.Bundle(dirs, ignore=True)
+
+    if all_:
+        res = list(dirs.tags.all)
+    else:
+        res = list(dirs.tags.any)
+
+    if has:
+        res = [tag in res for tag in has]
+
+    if res:
+        if json_:
+            click.echo(json.dumps(res, indent=4))
+        else:
+            for i in res:
+                click.echo(i)
+
+
+@cli.command()
+@click.argument("dirs", nargs=-1)
+@click.option("--all/--any", 'all_', default=True,
+              help="results for keys present in ANY treants")
+@click.option("--json", '-j', 'json_', is_flag=True,
+              help="results in JSON format")
+@click.option("--get", cls=OptionEatAll, help="get values for keys")
+def categories(dirs, all_, json_, get):
+    """Show treant categories"""
+    dirs = _handle_stdin(dirs)
+
+    # get all existing Treants among dirs
+    dirs = dtr.Bundle(dirs, ignore=True)
+
+    if get:
+        res = dirs.categories[set(get)]
+        keys = get
+    else:
+        if all_:
+            res = dict(dirs.categories.all)
+        else:
+            res = dict(dirs.categories.any)
+        keys = res.keys()
+
+    if res:
+        if json_:
+            click.echo(json.dumps(res, indent=4))
+        else:
+            for i, key in enumerate(keys):
+                for value in res[key]:
+                    click.echo("{} : {}".format(key, value))
+
+                if i < len(res) - 1:
+                    click.echo("")
+
+
+def _parse_categories(catstrings):
+    return {key: value for key, value in
```

Comment:
this was previously `json.loads(value)`, but that didn't work with eg `dtr init -c a:b`